src/odemis/util/peak.py

```diff
@@ -255,11 +293,15 @@ def _DoFit(self, future, spectrum, wavelength, type='gaussian'):
 
                 fit_list = []
                 for (pos, amplitude) in peaks:
-                    fit_list.append(pos)
-                    fit_list.append(width)
-                    fit_list.append(amplitude)
-                # Initialize offset to 0
-                fit_list.append(0)
+                    if type in {'gaussian_energy', 'lorentzian_energy'}:
+                        fit_list.extend(peak_to_energy(pos, width, amplitude))
+                    else:
+                        peak_params = [pos, width, amplitude]
+                        fit_list.extend(peak_params)
+
+                # Initialize offset to zero
```

Comment:
It's pretty clear that it's initialised to 0. A more interesting comment would be to indicate why not other values. In particular, we could mention that we've tried to use the minimum value, but that tended to lead the fitting to prefer having a large offset instead of smaller peaks, especially in energy domain.